src/citrine/_serialization/properties.py

```diff
@@ -175,22 +229,21 @@ def _set_elements(self, value: typing.Union[SerializedType, DeserializedType]):
         """
 
 
-def _get_base_class(obj: object, key: str) -> type:
+@lru_cache(maxsize=1024 * 1024)
+def _get_key_and_base_class(prop: Property, klass: typing.Any) -> \
+        typing.Tuple[typing.Optional[str], typing.Optional[str]]:
```

Comment:
Previously, we were incorrectly searching for serialization_path.  This first finds the actual field key and then resolves the MRO base class (instead of demanding that the property exist along exactly one inheritance path).